docs/proposals/algorithms/lifelong-learning/Implementation of a Class Incremental Learning Algorithm Evaluation System based on Ianvs.md

```diff
@@ -0,0 +1,226 @@
+
+In this context, it is necessary to develop an algorithm evaluation system that provides standardized testing for class-incremental learning algorithms, which is increasingly widely used in the industry, and evaluates the effectiveness of these algorithms.
+
+[KubeEdge-Ianvs](https://github.com/kubeedge/ianvs) is a distributed collaborative AI benchmarking project which can perform benchmarks with respect to several types of paradigms (e.g. single-task learning, incremental learning, etc.). This project aims to leverage the benchmarking capabilities of Ianvs to develop an evaluation system for class-incremental learning algorithms, in order to fulfill the benchmarking requirements specific to this type of algorithm.
+
+### 1.2 Goals
+
+This project aims to build a benchmarking for class-incremental learning in domain shift scenario on KubeEdge-Ianvs, it includes:
+ - Reproduce the Multi-Domain Incremental Learning for Semantic Segmentation (MDIL-SS) algorithm proposed in the [WACV2022 paper](https://github.com/prachigarg23/MDIL-SS).
+ - Use three datasets (including Cityscapes, SYNTHIA, and the Cloud-Robotic dataset provided by KubeEdge SIG AI) to conduct benchmarking tests and generate a comprehensive test report (including rankings, time, algorithm name, dataset,  and test metrics, among other details).
+
+## 2 Proposal
+
+`Implementation of a Class Incremental Learning Algorithm Evaluation System based on Ianvs` taking MDIL-SS algorithm as an example, aims to test the performance of class-incremental learning models following benchmarking standards, to make the development more efficient and productive.
+
+The scope of the system includes
+
+- A test case for class-incremental learning semantic segmentation algorithms, in which a test report can be successfully generated following instructions.
+- Easy to expand, allowing users to seamlessly integrate existing algorithms into the system for testing.
+
+Targeting users include
+
+- Beginners: Familiarize with distributed synergy AI and lifelong learning, among other concepts.
+- Developers: Quickly integrate class-increment algorithms into Ianvs and test the performance for further optimization.
+
+
+## 3 Design Details
+
+### 3.1 Overall Design
+
+The following is the architecture diagram of this project system, and this project focuses on the `unknown task processing` module.
+
+Before entering this module, unknown tasks have been [detected](https://github.com/kubeedge/ianvs/tree/4ae10f0e5e1ab958e143b04fade4acc448009857/examples/scene-based-unknown-task-recognition/lifelong_learning_bench) and samples have been labeled by some means such as manual labeling. The core concern of this module is how to use unknown task samples (i.e., incremental class samples) to update the model.
+![MDIL-SS](images/OSPP_MDIL-SS_6.png) 
```

Comment:
For the first suggestion, it has been clarified in the figure. 
For the second question, uploading reasoning results reduces the amount of data that needs to be transferred and preserves data privacy, which is very important in actual production.

Reply:
Ok, as for the second problem, I can see that the module 5-1 identifies unseen tasks and transmits data to the cloud for processing

Reply:
To clarify the second problem: in the framework of this project, unseen samples are detected at the edge, uploaded to the cloud for labeling, and eventually used for unseen task processing (project-focused part). The reason for this design is that in many cases, the computing and storage capacity of the edge is not enough to support model training, so unseen samples need to be uploaded to the cloud for labeling and training. The cloud is significantly better than the edge in terms of computing performance, and it can label and process unknown samples more quickly and efficiently.

Reply:
Thank you for your explanation. I understand.